components/navigable-container/stoppers.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress Dependencies
+ */
+import { keycodes } from '@wordpress/utils';
+
+/**
+ * Module Constants
+ */
+const { UP, DOWN, LEFT, RIGHT, TAB } = keycodes;
+
+function stopBubbling( event ) {
+	// Prevents arrow key handlers bound to the document directly interfering
+	event.nativeEvent.stopImmediatePropagation();
+	event.stopPropagation();
+}
+
+function handleHorizontalArrows( event ) {
+	// Do not preventDefault if inside a textarea or input where the arrow keys should move inside it.
+	if ( ! [ 'textarea', 'input' ].indexOf( event.target.nodeName.toLowerCase() ) ) {
+		event.preventDefault();
+	}
+
+	stopBubbling( event );
+}
+
+function stopEvent( event ) {
+	event.preventDefault();
+	stopBubbling( event );
+}
+
+export function stopEventIfRequired( stopArrowEvents, stopTabEvents, event ) {
+	const { keyCode } = event;
+
+	if ( stopArrowEvents && [ LEFT, RIGHT ].indexOf( keyCode ) > -1 ) {
+		handleHorizontalArrows( event );
+	} else if ( stopArrowEvents && [ UP, DOWN ].indexOf( keyCode ) > -1 ) {
```

Comment:
Why `UP/DOWN` are different than `LEFT/RIGHT` both need to move cursor in inputs/textarea right?

Reply:
Yep. Just confused myself. Up and down you don't typically use in a single line input, but you certainly would in a textarea.